src/KafkaFlow.OpenTelemetry/OpenTelemetryConsumerEventsHandler.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace KafkaFlow.OpenTelemetry
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Text;
+    using System.Threading.Tasks;
+    using global::OpenTelemetry;
+    using global::OpenTelemetry.Context.Propagation;
+
+    internal static class OpenTelemetryConsumerEventsHandler
+    {
+        private const string ProcessString = "process";
+        private static readonly TextMapPropagator Propagator = Propagators.DefaultTextMapPropagator;
+
+        public static Task OnConsumeStarted(IMessageContext context)
+        {
+            try
+            {
+                var activityName = !string.IsNullOrEmpty(context?.ConsumerContext.Topic) ? $"{context?.ConsumerContext.Topic} {ProcessString}" : ProcessString;
+
+                // Extract the PropagationContext of the upstream parent from the message headers.
+                var parentContext = Propagator.Extract(new PropagationContext(default, Baggage.Current), context, ExtractTraceContextIntoBasicProperties);
+                Baggage.Current = parentContext.Baggage;
+
+                // Start an activity with a name following the semantic convention of the OpenTelemetry messaging specification.
+                // The convention also defines a set of attributes (in .NET they are mapped as `tags`) to be populated in the activity.
+                // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md
+                var activity = KafkaFlowActivitySourceHelper.ActivitySource.StartActivity(activityName, ActivityKind.Consumer, parentContext.ActivityContext);
+
```

Comment:
``` csharp
if(activity is null)
{
    return Task.CompletedTask;
}
```
Does it make sense to add this code here?

Reply:
No. Even if the activity created here is null, we want the (possible) previous context to be propagated so we can avoid a broken trace.

Reply:
Looking at the code you just need to add the activity to the Items before the if. Doing that will prevent fewer null checks on the `activity`